File: articles/the-history-of-data-structures-and-algorithm-analysis/typescript/growth_rates.ts
for (const n of [10, 100, 1000, 10000]) {
  const log2 = Math.log2(n);
  console.log(`n=${n}, log2=${log2.toFixed(6)}, nlogn=${(n * log2).toFixed(6)}, n2=${n * n}`);
}

File: articles/the-history-of-data-structures-and-algorithm-analysis/java/GrowthRates.java
public class GrowthRates {
    public static void main(String[] args) {
        double[] ns = {10, 100, 1000, 10000};
        for (double n : ns) {
            System.out.printf("n=%.0f, log2=%.6f, nlogn=%.6f, n2=%.0f%n", n, Math.log(n) / Math.log(2), n * Math.log(n) / Math.log(2), n * n);
        }
    }
}
